### PR TITLE
feature/renderer-optmizations

### LINKED ISSUE #40: Any plans on focus system?
Hello, nice to see some open source projects in this area. Working right now with React and smart tv:s and built to multiple smart tv platforms and consoles (playstation). If I have time over I can see if I can contribute. One thing that I always thinks of is the focus system. Web is really good at mouse interaction but keyboard navigation is lacking. Is that something you are planning to look into?

Best regards

### PR DESCRIPTION
**Pull request by raphamorim**
**Current Status:** a lot of breaking changes!

# CLI

- [ ] Use `node-webos` as core CLI Builder ( shouldbeclose #67 )
- [ ] Install WebOS SDK based on Docker Linux with VNC  ( shouldbeclose #48 )

# Renderer
 
- [ ] Use `React.Reconciler` sync schedule as `false` by default (shouldbeclose #60)
- [ ] Spatial Navigation (closes #40 #63)
  - [x] `focusable` prop
  - [x] `focused` prop
  - [x] `onBlur`
  - [x] `onFocus`
  - [x] `onClick`
  - [ ] `onPress`

# All

- [x] Remove Benchmark Example
- [x] Add Navigation Example
- [x] Update to `yarn@1.3.2`, `node@>=8`, `npm@>=5`

# Tests

- [x] Test event handlers
- [ ] Test spatial navigation
- [ ] Test `node-webos` integration
- [ ] Test event handlers (quantity) after `updateContainer`

### CODE REVIEW

**Comment by raphamorim** [comment]
closes #60

File: src/renderer/events/Navigation.js [hunk 1]
```diff
@@ -0,0 +1,1193 @@
...
+  };
+  rect.center.left = rect.center.right = rect.center.x;
+  rect.center.top = rect.center.bottom = rect.center.y;
+  return rect;
+}
+
+function partition(rects, targetRect, straightOverlapThreshold) {
+  var groups = [[], [], [], [], [], [], [], [], []];
+
+  for (var i = 0; i < rects.length; i++) {
```
Comment (gkal19) [inline src/renderer/events/Navigation.js]: What do you think of making the following change from line 90 to line 94?
It's just a basic refactoring
```js
  for (const rect of rects) {
    const center = rect.center;
    let x;
    let y;
    let groupId;
```

Comment (raphamorim) [reply src/renderer/events/Navigation.js]: This code is borrowed from github.com/luke-chang/js-spatial-navigation
Before refactor it, I need to write some tests

**Comment by raphamorim** [comment]
I'll work on it in separate PR. It already is big enough.

**Pull request merged**